docs/admonitions.rst

```diff
@@ -2,219 +2,309 @@
+      class name to the rendered HTML elements.
+
+      .. rst-example::
+
+         .. admonition::
+            :class: annimated-admonition-border
+
+            .. literalinclude:: _static/extra_css.css
+               :language: css
+               :caption: docs/_static/extra_css.css
+               :start-after: /* ************************* annimated-admonition-border style
+               :end-before: /* ************************* my-special-key style
+
+Custom Admonitions
 ******************
 
-If you want to add a custom admonition type, all you need is a color and an \*.svg icon.
-Copy the icon's code from the `.icons <https://github.com/squidfunk/mkdocs-material/tree/master/material/.icons>`_
-folder and add the new CSS to an additional style sheet.
+This theme offers a robust solution that allows user-defined custom admonitions from the
+documentation's conf.py.
+
+.. confval:: sphinx_immaterial_custom_admonitions
+
+   A `list` of `dict`\ s that will be used to create custom admonition directives. Each `dict`
+   is validated using the data class `CustomAdmonitionConfig`.
 
-.. md-tab-set::
+   .. autoclass:: sphinx_immaterial.custom_admonitions.CustomAdmonitionConfig
+      :exclude-members: __new__
 
-   .. md-tab-item:: rST code
+Custom Admonition Example
+-------------------------
 
-      .. rst-example:: Pied Piper Example
-         :output-prefix:
+As a demonstration, we will be using the following configuration:
 
-         .. admonition:: Pied Piper
-            :class: pied-piper
+.. literalinclude:: conf.py
+   :language: python
+   :start-after: # BEGIN CUSTOM ADMONITIONS
+   :end-before: # END CUSTOM ADMONITIONS
+   :name: custom-admonition-example-config
 
-            Don't tell him you use spaces instead of tabs...
+Note that the name of the created directive (:rst:dir:`example-admonition`) is directly related to
+the value of the
+:py:obj:`~sphinx_immaterial.custom_admonitions.CustomAdmonitionConfig.__init__.name` option.
 
-   .. md-tab-item:: CSS code
+The above configuration will create a directive that could be documented like so:
 
-      .. literalinclude:: _static/extra_css.css
-         :language: css
-         :caption: docs/_static/extra_css.css
-         :start-after: /* *************************** custom admonition style rules
-         :end-before: /* **********
+.. rst:directive:: example-admonition
 
-   .. md-tab-item:: conf.py code
+   A custom admonition created from the
+   `example's configuration <custom-admonition-example-config>`.
+   See the :rst:dir:`admonition` directive for all available options and examples.
 
-      .. code-block:: python
-         :caption: docs/conf.py
+   .. rst-example::
 
-         html_static_path = ["_static"]
-         html_css_files = ["extra_css.css"]
+      .. example-admonition::
+
+         This is simple a example.
+
+
+.. example:: Legacy approach inherited from the mkdocs-material theme.
```

Comment:
Perhaps we don't need this example anymore?

Reply:
I left it in a collapsed admonition because it will still work. Same reason I left the section below this about changing the default icon (but with a warning about HTML bloat).

I didn't mention anything about the CSS classes `inline` and `end` inherited from upstream. They seem buggy when I use them, because the HTML structure (sphinx vs mkdocs) seems different enough to cause a bunch of alignment problems.

Reply:
I moved all inherited approaches into a separate section titled "Legacy Approaches", and I gave examples for how to migrate to the custom admonitions feature.